### PR TITLE
add support for custom success codes

### PR DESCRIPTION
**Pull request by thibaultmange**
Currently we can only return 200 success codes.
Adds possibility to send other types of success codes while remaining backward compatible.

### CODE REVIEW

File: jsonrest.go [hunk 1]
```diff
@@ -102,6 +102,11 @@ func (r *Request) URL() *url.URL {
 	return r.req.URL
 }
 
+type Response struct {
```
Comment (OSadovy) [inline jsonrest.go:105]: Can we please add a doc string to the exported symbol?

**Review by OSadovy — COMMENTED** [review]
headsup @johngibb  please take a look.


@thibaultmange  Should we also add a few lines to the doc explaining this new feature here? https://github.com/deliveroo/jsonrest-go/blob/16b3fe8630bbab788e44d181199b0154e5d958c8/doc.go#L8

File: jsonrest.go (hunk 1 above)
Comment (thibaultmange) [reply jsonrest.go:107]: sure, forgot it. Done

**Comment by thibaultmange** [comment]
Good point @OSadovy , updated the doc

**Review by OSadovy — APPROVED** [review]

**Pull request merged**